packages/ember-principled-forms/addon/components/form-field/component.ts

```diff
@@ -87,25 +89,26 @@ export default class FormField<T> extends Component {
     return !this.isInvalid ? undefined : 'true';
   }
 
-  // DISCUSS: What other useful defaults can we apply?
-  // e.g. an ARIA object
-  id: string = defaultTo(this.id, `form-field-${this.model.type}-${this.label}`);
-  errorId: string = defaultTo(
-    `${this.id}-error`,
-    `form-field-${this.model.type}-${this.label}-error`
-  );
-
   @computed('isInvalid')
   get ariaDescribedBy(): HTMLAttribute {
     return this.isInvalid ? this.errorId : undefined;
   }
 
-  onChange: (newValue: T) => void = defaultTo(this.onChange, noop);
-  onInput: (newValue: T) => void = defaultTo(this.onInput, noop);
+  onChange!: (newValue: T) => void;
+  onInput!: (newValue: T) => void;
 
-  constructor() {
-    super(...arguments);
+  init() {
+    super.init();
 
+    // DISCUSS: What other useful defaults can we apply?
+    // e.g. an ARIA object
+    this.id = defaultTo(this.id, `form-field-${this.model.type}-${this.label}`);
+    this.errorId = defaultTo(
```

Comment:
Note that given how `defaultTo` works, this will never actually be anything *but* `<this.id>-error`: `defaultTo` always uses the first argument if it's not `null` or `undefined`. (I don't recall when this got added originally, but clearly we missed it at that point.)